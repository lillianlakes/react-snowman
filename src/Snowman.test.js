import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";

it('does not allow players to keep guessing after the game is over', function() {
  const { container } = render(<Snowman />);
  // click on 7 buttons where letters are not in 'apple' so 7 wrong answers
  // fireEvent.click(container.querySelector('button[key="b"]'));
  // fireEvent.click(container.querySelector('button[key="c"]'));
  // fireEvent.click(container.querySelector('button[key="d"]'));
  // fireEvent.click(container.querySelector('button[key="f"]'));
  // fireEvent.click(container.querySelector('button[key="g"]'));
  // fireEvent.click(container.querySelector('button[key="h"]'));
  // fireEvent.click(container.querySelector('button[key="i"]'));
  
  fireEvent.click(container.querySelectorAll('button')[1]);
  fireEvent.click(container.querySelectorAll('button')[2]);
  fireEvent.click(container.querySelectorAll('button')[3]);
  fireEvent.click(container.querySelectorAll('button')[5]);
  fireEvent.click(container.querySelectorAll('button')[6]);
  fireEvent.click(container.querySelectorAll('button')[7]);
  // fireEvent.click(container.querySelectorAll('button')[8]);

  expect(container.querySelector('button')).toHaveClass("hidden");
  expect(container).toContainHTML("You lose");

 
})

it("matches snapshot", function() {
  const { container } = render(<Snowman />);
  const tooManyWrongGuesses = new Set();
  const alpha = "abcdefghijklmnopqrstuvwxyz";

  for (let i = 0; tooManyWrongGuesses.size <= Snowman.defaultProps.maxWrong + 1; i++ ) {
    if (!"apple".includes(alpha[i])) {
      tooManyWrongGuesses.add(alpha[i]);
    }
  } 
  
  Snowman.setGuessedLetters(tooManyWrongGuesses);
  
  expect(container).toMatchSnapshot();
});