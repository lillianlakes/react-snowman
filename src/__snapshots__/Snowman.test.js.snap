// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p>
      <button
        class="hidden"
        value="a"
      >
        a
      </button>
      <button
        class="hidden"
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        class="hidden"
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        class="hidden"
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        class="hidden"
        value="e"
      >
        e
      </button>
      <button
        class="hidden"
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        class="hidden"
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        class="hidden"
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        class="hidden"
        value="i"
      >
        i
      </button>
      <button
        class="hidden"
        value="j"
      >
        j
      </button>
      <button
        class="hidden"
        value="k"
      >
        k
      </button>
      <button
        class="hidden"
        value="l"
      >
        l
      </button>
      <button
        class="hidden"
        value="m"
      >
        m
      </button>
      <button
        class="hidden"
        value="n"
      >
        n
      </button>
      <button
        class="hidden"
        value="o"
      >
        o
      </button>
      <button
        class="hidden"
        value="p"
      >
        p
      </button>
      <button
        class="hidden"
        value="q"
      >
        q
      </button>
      <button
        class="hidden"
        value="r"
      >
        r
      </button>
      <button
        class="hidden"
        value="s"
      >
        s
      </button>
      <button
        class="hidden"
        value="t"
      >
        t
      </button>
      <button
        class="hidden"
        value="u"
      >
        u
      </button>
      <button
        class="hidden"
        value="v"
      >
        v
      </button>
      <button
        class="hidden"
        value="w"
      >
        w
      </button>
      <button
        class="hidden"
        value="x"
      >
        x
      </button>
      <button
        class="hidden"
        value="y"
      >
        y
      </button>
      <button
        class="hidden"
        value="z"
      >
        z
      </button>
    </p>
    <p
      class=""
    >
      You lose. Correct Word was: "apples"
    </p>
  </div>
</div>
`;
